### PR TITLE
Update Examples for Release 3.2.1

### PR DESCRIPTION
**Pull request by awholegunch**
**This is a rather large PR, in that every example was modified, and in some ways extensively.**

There was a lot of repetitive text in the examples, explaining how to start and stop flows, services, workflows, etc. Instead of having the same copy repeated in multiple rst files, requiring complicated and careful changes with search 'n replace, the repetitive material has—as much as easily possible—been moved to included files, and an pseudo-API created to allow each example to specified. Unfortunately, two of our examples (Purchase and StreamConversion) don't follow the naming convention of the other examples where the directory and the Application are the same, making things more complicated. Perhaps a future PR will change that.

These examples have all been tested (manually) on the Mac against the 3.2.0 SDK. Originally, this effort was going to released immediately, but as 3.2.1 is soon upon us, I would rather finish this up and re-test against 3.2.1.

It updates the screenshots and commands for recent changes in the UI. It fixes multiple mistakes in the example commands, descriptions, and adds new commentary where this correspondent found they lacked clarity. The directory ``_includes`` contains common boilerplate that is included in different files. 

One change that was **not** included but will be addressed in a different PR is that of 
https://issues.cask.co/browse/CDAP-3949; I found that the SDK needed to have 2048 in memory to run all the examples successfully, especially Wikipedia-Pipeline.

See: https://github.com/caskdata/cdap/pull/4280

**Reviewing this is going to be a challenge**, as the changes are so many. However, there are 15 examples; if five people took three each, and followed the rendered text rather than the rst, try the example and make sure they work, then it would go faster. **You could sign up by leaving a comment on this PR as to which example(s) you are going to test and check.**

The changes pass the [Quick Build](http://builds.cask.co/browse/CDAP-DOB12-2); here is [the new examples index page](http://builds.cask.co/artifact/CDAP-DOB12/shared/build-2/Docs-HTML/3.2.0/en/examples-manual/examples/index.html).

Addresses issue https://issues.cask.co/browse/CDAP-3910.

### CODE REVIEW

File: cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst [hunk 1]
```diff
@@ -11,40 +11,50 @@ Wikipedia Pipeline
 
 A Cask Data Application Platform (CDAP) example demonstrating a typical data processing pipeline using CDAP Workflows.
```
Comment (bdmogal) [inline cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst]: not this PR, but ... a typical **batch** data processing pipeline...

File: cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst [hunk 2]
```diff
@@ -11,40 +11,50 @@ Wikipedia Pipeline
...
 Once raw Wikipedia data is available from using either the online or offline modes, the
 *WikipediaPipelineWorkflow* runs a MapReduce program *WikiContentValidatorAndNormalizer* that filters bad records from
 the raw data, as well as normalizes it by converting the wikitext-formatted data to plain text. It then stores the
-output in another KeyValueTable dataset *normalized*.
+output in another ``KeyValueTable`` dataset called *normalized*.
+
+The *WikipediaPipelineWorkflow* then contains a fork, with two branches:
+
+- One branch runs the Apache Spark program *SparkWikipediaAnalyzer*. This program consumes
+  normalized data and runs topic modeling on it using the Latent Dirichlet Allocation (LDA)
```
Comment (bdmogal) [inline cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst]: perhaps a link for LDA? https://en.wikipedia.org/wiki/Latent_Dirichlet_allocation

File: cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst [hunk 3]
```diff
@@ -84,111 +101,100 @@ The *pageTitleStream* consumes events in the format returned by the Facebook "Li
...
+Reduce the number of minimum number of pages to zero, and change the mode to *online*, by
+setting either the runtime arguments (which changes the *next* run) or the preferences
+(which changes *all* subsequent runs):
 
-The *WikipediaService* exposes these REST APIs:
+.. image:: _images/wikipedia-data-pipeline-2.png
+   :width: 8in
 
-- Retrieve the list of topics generated by the *SparkWikipediaAnalyzer* program::
+Once the pipeline has run through to the end, you can start the service and query the results.
```
Comment (bdmogal) [inline cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst]: for the sake of completeness, should we also add an image that shows a DAG after the workflow has run through to the end?

**Comment by bdmogal** [comment]
@awholegunch only minor comments about ``WikipediaPipeline``

File: cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst (hunk 2 above)
Comment (awholegunch) [inline cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst]: added

File: cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst (hunk 1 above)
Comment (awholegunch) [inline cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst]: fixed.

File: cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst (hunk 3 above)
Comment (awholegunch) [inline cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst]: added.

**Comment by awholegunch** [comment]
@bdmogal [Updated Wikipedia Pipeline Example](http://builds.cask.co/artifact/CDAP-DOB12/shared/build-3/Docs-HTML/3.2.0/en/examples-manual/examples/wikipedia-data-pipeline.html)

**Comment by sreevatsanraman** [comment]
LGTM :+1:

**Pull request merged**